compiler/src/dotty/tools/dotc/quoted/reflect/ReflectionCompilerInterface.scala

```diff
@@ -1335,7 +1335,10 @@ class ReflectionCompilerInterface(val rootContext: Context) extends CompilerInte
   def AppliedType_tycon(self: AppliedType)(using Context): Type = self.tycon
   def AppliedType_args(self: AppliedType)(using Context): List[TypeOrBounds] = self.args
 
-  def AppliedType_apply(tycon: Type, args: List[TypeOrBounds])(using Context): AppliedType = Types.AppliedType(tycon, args)
+  def AppliedType_apply(tycon: Type, args: List[TypeOrBounds])(using Context): Type =
+    tycon match
```

Comment:
I think the universal thing you want is tycon.appliedTo, but then I would just expose this method and remove the AppliedType_apply

Reply:
I also changed the `args` to a `List[Type]` for applied types in general as we cannot have `NoPrefix` or `TypeBounds` as arguments.

Reply:
How do I write down `java.util.List[_ <: Object]` if I can't use TypeBounds as arguments?

Reply:
I will revert it.

Reply:
Done